Validate `ignore` and `only` compiletest directive, and add human-readable ignore reasons

emilyalbini opened the pull request:
This PR adds strict validation for the `ignore` and `only` compiletest directives, failing if an unknown value is provided to them. Doing so uncovered 79 tests in `tests/ui` that had invalid directives, so this PR also fixes them.

Finally, this PR adds human-readable ignore reasons when tests are ignored due to `ignore` or `only` directives, like *"only executed when the architecture is aarch64"* or *"ignored when the operative system is windows"*. This was the original reason why I started working on this PR and #108659, as we need both of them for Ferrocene.

The PR is a draft because the code is extremely inefficient: it calls `rustc --print=cfg --target $target` for every rustc target (to gather the list of allowed ignore values), which on my system takes between 4s and 5s, and performs a lot of allocations of constant values. I'll fix both of them in the coming days.

r? @ehuss

emilyalbini committed:
include eventual comment in the compiletest ignore reason

emilyalbini committed:
include pretty reason why ignore-FOO matched

emilyalbini committed:
fix ignore-thumbv8m.base

emilyalbini committed:
remove invalid ignore-pretty

emilyalbini committed:
remove invalid ignore-powerpc64le

emilyalbini committed:
split invalid only-x86-windows into only-x86 and only-windows

emilyalbini committed:
validate ignore-FOO/only-FOO directives and output only-FOO reasoning

emilyalbini committed:
reduce boilerplate with common enums

bjorn3 commented on tests/codegen/global_asm.rs:
```diff
@@ -10,7 +10,6 @@
 // ignore-mips64
 // ignore-msp430
 // ignore-powerpc64
-// ignore-powerpc64le
```
There are several targets starting with `powerpc64le`:

```
powerpc64le-unknown-freebsd
powerpc64le-unknown-linux-gnu
powerpc64le-unknown-linux-musl
```

Is `ignore-powerpc64` enough to match them?

pietroalbini replied:
`ignore-powerpc64le` did not match anything. Both before and after this PR, unless the full target was specified (or the end of the target, even though that was not used anywhere), the directive looked at the `cfg`s:

```
$ rustc --print=cfg --target powerpc64le-unknown-linux-gnu
debug_assertions
panic="unwind"
target_arch="powerpc64"
target_endian="little"
target_env="gnu"
target_family="unix"
target_has_atomic="16"
target_has_atomic="32"
target_has_atomic="64"
target_has_atomic="8"
target_has_atomic="ptr"
target_os="linux"
target_pointer_width="64"
target_vendor="unknown"
unix
```

So, in this case, it would match `ignore-powerpc64`, as that's what is in the `target_arch`.

bjorn3 commented on tests/ui/symbol-names/x86-stdcall.rs:
```diff
@@ -1,5 +1,6 @@
 // build-pass
-// only-x86-windows
+// only-x86
+// only-windows
```
Nice catch!

Mark-Simulacrum commented on tests/ui/asm/issue-87802.rs:
```diff
@@ -1,6 +1,5 @@
 // needs-asm-support
 // ignore-nvptx64
-// ignore-spirv
```
I think we should treat this similarly to check-cfg (maybe we can even lean on that?) -- https://github.com/rust-lang/rust/blob/master/src/bootstrap/lib.rs#L127-L130 has a list of extra things we allow.

pietroalbini replied:
Fixed.

emilyalbini committed:
implement --print=all-target-specs-json

emilyalbini committed:
use target specs rather than --print=cfg to discover targets

emilyalbini committed:
use --print=all-target-specs-json for stage1+

pietroalbini commented:
Ok so with `--print all-target-specs-json` it's now way faster again. Tomorrow I'll look into the test failure, handle `""` being inserted into the hashsets, and possibly look if I can speed startup up even more.

ehuss commented on compiler/rustc_driver_impl/src/lib.rs:
```diff
@@ -644,6 +646,15 @@ fn print_crate_info(
+                let mut targets = BTreeMap::new();
+                for name in rustc_target::spec::TARGETS.to_vec() {
+                    let triple = TargetTriple::from_triple(name);
+                    let Ok((target, _)) = Target::search(&triple, &sess.sysroot) else { continue };
```
Is there a particular reason this uses `search` instead of loading it directly, like:

```suggestion
                    let target = Target::expect_builtin(&triple);
```

pietroalbini replied:
Not really, not much familiar with this part of the codebase and that was the method I found doing a brief search.

ehuss commented on compiler/rustc_driver_impl/src/lib.rs:
```diff
@@ -644,6 +646,15 @@ fn print_crate_info(
             }
+            AllTargetSpecs => {
+                let mut targets = BTreeMap::new();
+                for name in rustc_target::spec::TARGETS.to_vec() {
```
This `to_vec` seems unnecessary

```suggestion
                for name in rustc_target::spec::TARGETS {
```

pietroalbini replied:
...right!

ehuss reviewed (commented):
This looks great, thanks for putting it together!

emilyalbini committed:
avoid cloning the whole compiletest configuration for every test

emilyalbini committed:
handle "ignore-" and "only-"

emilyalbini committed:
move cfg handling into its own module

emilyalbini committed:
reduce allocations when validating cfgs

emilyalbini committed:
allow some out of tree archs

emilyalbini committed:
migrate existing behavior of matches_arch

emilyalbini committed:
properly match multiple families

emilyalbini committed:
fix remaining tests

ehuss commented:
Looks like it needs to be formatted.

Also, I assume the 10 day period for https://github.com/rust-lang/compiler-team/issues/600 needs to finish?

bors commented:
:umbrella: The latest upstream changes (presumably #109097) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

pietroalbini commented:
Rebased.

lqd commented on tests/ui/panic-runtime/unwind-tables-target-required.rs:
```diff
@@ -1,7 +1,7 @@
 // when they are required.
 //
-// only-x86_64-windows-msvc
+// only-x86_64-pc-windows-msvc
```
note that I already did this one in https://github.com/rust-lang/rust/pull/108809 and it was not enough

emilyalbini committed:
add support for ignore-llvm-version

emilyalbini committed:
remove a bunch of unknown archs from the global_asm tests

emilyalbini committed:
make 32bit ignore more accurate

emilyalbini committed:
remove unknown xcore arch

emilyalbini committed:
add support for ignore-gdb-version

emilyalbini committed:
restore check for both target os and env

This is better than the old impl of target.ends_with("windows-gnu"),
because it also catches things like windows-gnullvm

emilyalbini committed:
fix wrong ignore condition

emilyalbini committed:
fix solaris ignore

pietroalbini commented:
Pushed changes to fix the broken annotations in other compiletest suites (in addition to `tests/ui`).

bors commented:
:umbrella: The latest upstream changes (presumably #108659) made this pull request unmergeable. Please [resolve the merge conflicts](https://rustc-dev-guide.rust-lang.org/git.html#rebasing-and-conflicts).

ehuss commented on src/tools/compiletest/src/header/tests.rs:
```diff
@@ -430,6 +430,8 @@ fn wasm_special() {
     for (target, pattern, ignore) in ignores {
         let mut config = config();
         config.target = target.to_string();
+        dbg!(target, pattern, ignore);
+        dbg!(config.target_cfgs());
```
You may want to remove these since they generate a large amount of output.

pietroalbini replied:
Whelp that wasn't supposed to be left there, I'll remove it tomorrow.

ehuss reviewed (approved):
r=me with rebase 

I haven't done a thorough review to check which tests will start or stop running with our CI configurations, or that this won't pass/fail on some of the more unusual targets. However, the changes look valid to me on the surface.

Perhaps someone could review the ignore messages in CI, and compare to a previous run, but I think that will be a lot of work.

pietroalbini commented:
I'll rebase tomorrow!

> Perhaps someone could review the ignore messages in CI, and compare to a previous run, but I think that will be a lot of work.

Well, good thing that all the executed tests are now available in the build metrics JSON files since #108659 :joy:

pietroalbini commented:
"tomorrow"

@bors r=ehuss rollup=never

bors commented:
:pushpin: Commit 4fb3e65fc436cf70ae995dcb20d148492daa4d02 has been approved by `ehuss`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ehuss 4fb3e65fc436cf70ae995dcb20d148492daa4d02 -->
<!-- homu: {"type":"Approved","sha":"4fb3e65fc436cf70ae995dcb20d148492daa4d02","approver":"ehuss","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

rust-log-analyzer commented:
The job **`armhf-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4562257859/jobs/8049228304) [(plain)](https://github.com/rust-lang-ci/rust/commit/9993a8f192323b98716fd4289df1d85f4098d5f4/checks/12388780705/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [ui] tests/ui/coercion/coerce-block-tail-26978.rs ... ok
test [ui] tests/ui/codemap_tests/unicode_3.rs ... ok
test [ui] tests/ui/coercion/coerce-block-tail-57749.rs ... ok
test [ui] tests/ui/coercion/coerce-block-tail-83783.rs ... ok
test [ui] tests/ui/coercion/coerce-issue-49593-box-never-windows.rs#fallback ... ignored, only executed when the operative system is windows (the number of `Error` impls is platform-dependent)
test [ui] tests/ui/coercion/coerce-issue-49593-box-never-windows.rs#nofallback ... ignored, only executed when the operative system is windows (the number of `Error` impls is platform-dependent)
test [ui] tests/ui/coercion/coerce-expect-unsized-ascribed.rs ... ok
test [ui] tests/ui/coercion/coerce-block-tail-83850.rs ... ok
test [ui] tests/ui/coercion/coerce-issue-49593-box-never.rs#fallback ... ok
test [ui] tests/ui/coercion/coerce-issue-49593-box-never.rs#nofallback ... ok
---
test [ui] tests/ui/consts/offset_from.rs ... ok
test [ui] tests/ui/consts/static_mut_containing_mut_ref2.rs#mut_refs ... ok
test [ui] tests/ui/consts/static_mut_containing_mut_ref.rs ... ok
test [ui] tests/ui/consts/static_mut_containing_mut_ref2.rs#stock ... ok
test [ui] tests/ui/consts/std/alloc.rs ... ignored, ignored when building with debug assertions ((the debug assertions change the error))
test [ui] tests/ui/consts/packed_pattern.rs ... ok
test [ui] tests/ui/consts/std/cell.rs ... ok
test [ui] tests/ui/consts/packed_pattern2.rs ... ok
test [ui] tests/ui/consts/too_generic_eval_ice.rs ... ok
---
test [ui] tests/ui/invalid/invalid-inline.rs ... ok
test [ui] tests/ui/intrinsics/intrinsic-atomics-cc.rs ... ok
test [ui] tests/ui/invalid/invalid-macro-matcher.rs ... ok
test [ui] tests/ui/invalid/invalid-no-sanitize.rs ... ok
test [ui] tests/ui/io-checks/inaccessbile-temp-dir.rs ... ignored, ignored when the architecture is arm (the file-system issues do not replicate here, at least on armhf-gnu)
test [ui] tests/ui/io-checks/non-ice-error-on-worker-io-fail.rs ... ignored, ignored when the architecture is arm (the file-system issues do not replicate here, at least on armhf-gnu)
test [ui] tests/ui/invalid/invalid-plugin-attr.rs ... ok
test [ui] tests/ui/invalid/invalid-path-in-const.rs ... ok
test [ui] tests/ui/invalid/invalid-rustc_legacy_const_generics-arguments.rs ... ok
test [ui] tests/ui/invalid/invalid_rustc_layout_scalar_valid_range.rs ... ok
---
test [ui] tests/ui/issues/issue-6936.rs ... ok
test [ui] tests/ui/issues/issue-69396-const-no-type-in-macro.rs ... ok
test [ui] tests/ui/issues/issue-69455.rs ... ok
test [ui] tests/ui/issues/issue-69602-type-err-during-codegen-ice.rs ... ok
test [ui] tests/ui/issues/issue-70093/issue-70093-link-directives.rs ... ignored, ignored when cross-compiling (default-linker-libraries=yes doesn't play well with cross compiling)
test [ui] tests/ui/issues/issue-70093/issue-70093.rs ... ignored, ignored when cross-compiling (default-linker-libraries=yes doesn't play well with cross compiling)
test [ui] tests/ui/issues/issue-70381.rs ... ok
test [ui] tests/ui/issues/issue-61696.rs ... ok
test [ui] tests/ui/issues/issue-7044.rs ... ok
test [ui] tests/ui/issues/issue-7061.rs ... ok
---
test [ui] tests/ui/iterators/iter-range.rs ... ok
test [ui] tests/ui/iterators/string.rs ... ok
test [ui] tests/ui/json/json-and-color.rs ... ok
test [ui] tests/ui/iterators/vec-on-unimplemented.rs ... ok
test [ui] tests/ui/json/json-bom-plus-crlf-multifile-aux.rs ... ignored, ignored always (Not a test. Used by other tests)
test [ui] tests/ui/json/json-bom-plus-crlf-multifile.rs ... ok
test [ui] tests/ui/json/json-bom-plus-crlf.rs ... ok
test [ui] tests/ui/json/json-invalid.rs ... ok
test [ui] tests/ui/json/json-multiple.rs ... ok
---
test [ui] tests/ui/lint/command-line-lint-group-deny.rs ... ok
test [ui] tests/ui/lint/command-line-register-lint-tool.rs ... ok
test [ui] tests/ui/lint/command-line-register-unknown-lint-tool.rs ... ok
test [ui] tests/ui/lint/command-line-lint-group-warn.rs ... ok
test [ui] tests/ui/lint/dead-code/closure-bang.rs ... ignored, ignored always (FIXME(#20574))
test [ui] tests/ui/lint/crate_level_only_lint.rs ... ok
test [ui] tests/ui/lint/dead-code/anon-const-in-pat.rs ... ok
test [ui] tests/ui/lint/clashing-extern-fn.rs ... ok
test [ui] tests/ui/lint/dead-code/basic.rs ... ok
---
test [ui] tests/ui/lint/dead-code/lint-dead-code-2.rs ... ok
test [ui] tests/ui/limits/huge-struct.rs ... ok
test [ui] tests/ui/lint/dead-code/lint-dead-code-3.rs ... ok
test [ui] tests/ui/lint/dead-code/lint-dead-code-4.rs ... ok
test [ui] tests/ui/lint/dead-code/self-assign.rs ... ignored, ignored always (FIXME(81658, 83171))
test [ui] tests/ui/lint/dead-code/lint-dead-code-6.rs ... ok
test [ui] tests/ui/lint/dead-code/multiple-dead-codes-in-the-same-struct.rs ... ok
test [ui] tests/ui/lint/dead-code/newline-span.rs ... ok
test [ui] tests/ui/lint/dead-code/trait-impl.rs ... ok
test [ui] tests/ui/lint/dead-code/trait-impl.rs ... ok
test [ui] tests/ui/lint/dead-code/tuple-struct-field.rs ... ok
test [ui] tests/ui/lint/dead-code/type-alias.rs ... ok
test [ui] tests/ui/lint/dead-code/type-in-foreign.rs ... ok
test [ui] tests/ui/lint/dead-code/unused-enum.rs ... ok
test [ui] tests/ui/lint/dead-code/unused-struct-variant.rs ... ok
test [ui] tests/ui/lint/dead-code/with-core-crate.rs ... ok
test [ui] tests/ui/lint/dead-code/unused-variant.rs ... ok
test [ui] tests/ui/lint/expansion-time-include.rs ... ignored, ignored always (auxiliary file for expansion-time.rs)
test [ui] tests/ui/lint/empty-lint-attributes.rs ... ok
test [ui] tests/ui/lint/expansion-time.rs ... ok
test [ui] tests/ui/lint/expr_attr_paren_order.rs ... ok
test [ui] tests/ui/lint/dead-code/unused-variant-pub.rs ... ok
---
test [ui] tests/ui/lto/all-crates.rs ... ok
test [ui] tests/ui/macros/format-parse-errors.rs ... ok
test [ui] tests/ui/macros/global-asm.rs ... ok
test [ui] tests/ui/macros/format-unused-lables.rs ... ok
test [ui] tests/ui/macros/include-single-expr-helper-1.rs ... ignored, ignored always (auxiliary file for include-single-expr.rs)
test [ui] tests/ui/macros/include-single-expr-helper.rs ... ignored, ignored always (auxiliary file for include-single-expr.rs)
test [ui] tests/ui/lto/fat-lto.rs ... ok
test [ui] tests/ui/macros/issue-102878.rs ... ok
test [ui] tests/ui/macros/issue-103529.rs ... ok
test [ui] tests/ui/macros/issue-104769-concat_bytes-invalid-literal.rs ... ok
---
test [ui] tests/ui/macros/issue-6596-1.rs ... ok
test [ui] tests/ui/macros/issue-63102.rs ... ok
test [ui] tests/ui/macros/issue-68058.rs ... ok
test [ui] tests/ui/macros/issue-25274.rs ... ok
test [ui] tests/ui/macros/issue-69838-dir/bar.rs ... ignored, ignored always (this is an auxiliary file as part of another test.)
test [ui] tests/ui/macros/issue-69838-dir/included.rs ... ignored, ignored always (this is an auxiliary file as part of another test.)
test [ui] tests/ui/macros/issue-33185.rs ... ok
test [ui] tests/ui/macros/issue-70446.rs ... ok
test [ui] tests/ui/macros/issue-69838-mods-relative-to-included-path.rs ... ok
test [ui] tests/ui/macros/issue-77475.rs ... ok
---
test [ui] tests/ui/missing/missing-derivable-attr.rs ... ok
test [ui] tests/ui/missing/missing-fields-in-struct-pattern.rs ... ok
test [ui] tests/ui/mismatched_types/suggest-removing-tuple-struct-field.rs ... ok
test [ui] tests/ui/mir/mir_temp_promotions.rs ... ok
test [ui] tests/ui/missing_non_modrs_mod/foo.rs ... ignored, ignored always (this is just a helper for the real test in this dir)
test [ui] tests/ui/missing_non_modrs_mod/foo_inline.rs ... ignored, ignored always (this is just a helper for the real test in this dir)
test [ui] tests/ui/missing-trait-bounds/issue-35677.rs ... ok
test [ui] tests/ui/missing-trait-bounds/missing-trait-bound-for-op.rs ... ok
test [ui] tests/ui/missing-trait-bounds/missing-trait-bounds-for-method-call.rs ... ok
test [ui] tests/ui/missing/missing-comma-in-match.rs ... ok
---
test [ui] tests/ui/module-macro_use-arguments.rs ... ok
test [ui] tests/ui/modules/issue-56411.rs ... ok
test [ui] tests/ui/mir/validate/needs-reveal-all.rs ... ok
test [ui] tests/ui/missing/missing-items/m2.rs ... ok
test [ui] tests/ui/modules/mod_file_aux.rs ... ignored, ignored always (Not a test. Used by other tests)
test [ui] tests/ui/missing/missing-macro-use.rs ... ok
test [ui] tests/ui/missing_debug_impls.rs ... ok
test [ui] tests/ui/modules/path-macro.rs ... ok
test [ui] tests/ui/modules/path-no-file-name.rs ... ok
test [ui] tests/ui/modules/path-no-file-name.rs ... ok
test [ui] tests/ui/modules/path-invalid-form.rs ... ok
test [ui] tests/ui/modules_and_files_visibility/mod_file_aux.rs ... ignored, ignored always (Not a test. Used by other tests)
test [ui] tests/ui/modules_and_files_visibility/mod_file_correct_spans.rs ... ok
test [ui] tests/ui/modules_and_files_visibility/mod_file_correct_spans.rs ... ok
test [ui] tests/ui/modules_and_files_visibility/mod_file_disambig_aux.rs ... ignored, ignored always (not a test. aux file)
test [ui] tests/ui/modules_and_files_visibility/mod_file_disambig_aux/mod.rs ... ignored, ignored always (not a test. aux file)
test [ui] tests/ui/missing-trait-bounds/issue-69725.rs ... ok
test [ui] tests/ui/moves/borrow-closures-instead-of-move.rs ... ok
test [ui] tests/ui/moves/issue-46099-move-in-macro.rs ... ok
test [ui] tests/ui/moves/issue-72649-uninit-in-loop.rs ... ok
---
test [ui] tests/ui/no-warn-on-field-replace-issue-34101.rs ... ok
test [ui] tests/ui/no_send-enum.rs ... ok
test [ui] tests/ui/no_send-rc.rs ... ok
test [ui] tests/ui/no_share-enum.rs ... ok
test [ui] tests/ui/non_modrs_mods/foors_mod.rs ... ignored, ignored always (not a test, used by non_modrs_mods.rs)
test [ui] tests/ui/non_modrs_mods_and_inline_mods/x.rs ... ignored, ignored always (not a test)
test [ui] tests/ui/non_modrs_mods_and_inline_mods/x/y/z/mod.rs ... ignored, ignored always (not a test)
test [ui] tests/ui/no_share-struct.rs ... ok
test [ui] tests/ui/no-reuse-move-arc.rs ... ok
---
test [ui] tests/ui/packed/issue-27060-2.rs ... ok
test [ui] tests/ui/overloaded/overloaded-autoderef-xcrate.rs ... ok
test [ui] tests/ui/overloaded/overloaded-calls-object-one-arg.rs ... ok
test [ui] tests/ui/packed/issue-27060.rs ... ok
test [ui] tests/ui/packed/packed-struct-borrow-element-64bit.rs ... ignored, ignored when the pointer width is 32bit ((needs `usize` to be 8-aligned to reproduce all the errors below))
test [ui] tests/ui/overloaded/overloaded-calls-object-two-args.rs ... ok
test [ui] tests/ui/overloaded/overloaded-calls-param-vtables.rs ... ok
test [ui] tests/ui/overloaded/overloaded-autoderef.rs ... ok
test [ui] tests/ui/overloaded/overloaded-index-in-field.rs ... ok
---
test [ui] tests/ui/parser/bounds-lifetime.rs ... ok
test [ui] tests/ui/parser/bounds-obj-parens.rs ... ok
test [ui] tests/ui/parser/byte-string-literals.rs ... ok
test [ui] tests/ui/parser/byte-literals.rs ... ok
test [ui] tests/ui/parser/circular_modules_hello.rs ... ignored, ignored always (this is an auxiliary file for circular-modules-main.rs)
test [ui] tests/ui/panics/runtime-switch.rs#legacy ... ok
test [ui] tests/ui/parser/class-implements-bad-trait.rs ... ok
test [ui] tests/ui/parser/closure-return-syntax.rs ... ok
test [ui] tests/ui/parser/char/whitespace-character-literal.rs ... ok
---
test [ui] tests/ui/parser/issues/issue-33413.rs ... ok
test [ui] tests/ui/parser/issues/issue-34255-1.rs ... ok
test [ui] tests/ui/parser/issues/issue-43196.rs ... ok
test [ui] tests/ui/parser/issues/issue-41155.rs ... ok
test [ui] tests/ui/parser/issues/issue-48508-aux.rs ... ignored, ignored always (Not a test. Used by issue-48508.rs)
test [ui] tests/ui/parser/issues/issue-33418.rs ... ok
test [ui] tests/ui/parser/issues/issue-43692.rs ... ok
test [ui] tests/ui/parser/issues/issue-48137-macros-cannot-interpolate-impl-items-bad-variants.rs ... ok
test [ui] tests/ui/parser/issues/issue-44406.rs ... ok
---
test [ui] tests/ui/rmeta/rmeta.rs ... ok
test [ui] tests/ui/rfcs/rfc-2005-default-binding-mode/tuple-struct.rs ... ok
test [ui] tests/ui/rfcs/rfc-2005-default-binding-mode/tuple.rs ... ok
test [ui] tests/ui/rmeta/rmeta_meta_main.rs ... ok
test [ui] tests/ui/runtime/backtrace-debuginfo-aux.rs ... ignored, ignored always (not a test, used by backtrace-debuginfo.rs to test file!())
test [ui] tests/ui/rfcs/rfc-2151-raw-identifiers/items.rs ... ok
test [ui] tests/ui/rmeta/rmeta_lib.rs ... ok
test [ui] tests/ui/rfcs/rfc-2302-self-struct-ctor.rs ... ok
test [ui] tests/ui/rfcs/rfc-2151-raw-identifiers/macros.rs ... ok
---
test [ui] tests/ui/type-alias-impl-trait/issue-60407.rs ... ok
test [ui] tests/ui/tuple/tup.rs ... ok
test [ui] tests/ui/type-alias-impl-trait/issue-60564-working.rs ... ok
test [ui] tests/ui/type-alias-impl-trait/issue-62000-associate-impl-trait-lifetimes.rs ... ok
test [ui] tests/ui/type-alias-impl-trait/issue-65918.rs ... ignored, ignored always (This now ICEs again.)
test [ui] tests/ui/type-alias-impl-trait/issue-60564.rs ... ok
test [ui] tests/ui/type-alias-impl-trait/issue-63355.rs ... ok
test [ui] tests/ui/type-alias-impl-trait/issue-65384.rs ... ok
test [ui] tests/ui/type-alias-impl-trait/issue-65679-inst-opaque-ty-from-val-twice.rs ... ok
---
---- [ui] tests/ui/regions/regions-bot.rs stdout ----

error: test run failed!
status: exit status: 101
command: RUST_TEST_THREADS="16" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/remote-test-client" "run" "0" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/regions/regions-bot/a"
--- stdout -------------------------------
uploaded "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/regions/regions-bot/a", waiting for result
--- stderr -------------------------------
thread 'main' panicked at 'client.read_exact(&mut header) failed with Connection reset by peer (os error 104)', src/tools/remote-test-client/src/main.rs:310:9
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace
------------------------------------------
```

</details>

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4562257859/jobs/8049749287)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4562257859/jobs/8049749287","builder_name":"checks-actions"} -->

bors commented:
:hourglass: Testing commit 4fb3e65fc436cf70ae995dcb20d148492daa4d02 with merge e4ce36f53a9f19216767e981b5605d43f2bd4be8...
<!-- homu: {"type":"BuildStarted","head_sha":"4fb3e65fc436cf70ae995dcb20d148492daa4d02","merge_sha":"e4ce36f53a9f19216767e981b5605d43f2bd4be8"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4562522032/jobs/8051239152)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4562522032/jobs/8051239152","builder_name":"checks-actions"} -->

pietroalbini commented:
Ok, so the `tests/ui/symbol-names/x86-stdcall.rs` test is broken at least on i686-mingw. @nbdd0121 you added that test in https://github.com/rust-lang/rust/pull/96444, is it expected for the test to fail on mingw, or is the test broken?

nbdd0121 commented:
It seems that vectorcall is not supported by GCC: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89485, and thus mingw linker cannot resolve the correctly mangled symbol.

I think disabling the test for mingw is sufficient.

emilyalbini committed:
ignore x86-stdcall on mingw

pietroalbini commented:
Thanks! Ignored the test on `-gnu`.

@bors r=ehuss

bors commented:
:pushpin: Commit b7f6ee55152ba04bb8b2e64b9e622bc7350add99 has been approved by `ehuss`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ehuss b7f6ee55152ba04bb8b2e64b9e622bc7350add99 -->
<!-- homu: {"type":"Approved","sha":"b7f6ee55152ba04bb8b2e64b9e622bc7350add99","approver":"ehuss","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

rust-log-analyzer commented:
The job **`i686-mingw-2`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4562522032/jobs/8049863946) [(plain)](https://github.com/rust-lang-ci/rust/commit/e4ce36f53a9f19216767e981b5605d43f2bd4be8/checks/12389597115/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [ui] tests\ui\coherence\impl[t]-foreign[t]-for-fundamental.rs ... ok
test [ui] tests\ui\coinduction\canonicalization-rerun.rs#new ... ok
test [ui] tests\ui\coherence\issue-85026.rs ... ok
test [ui] tests\ui\coherence\impl[t]-foreign[local]-for-t.rs ... ok
test [ui] tests\ui\command\command-argv0.rs ... ignored, ignored when the operative system is windows (this is a unix-specific test)
test [ui] tests\ui\command\command-create-pidfd.rs ... ignored, only executed when the operative system is linux (pidfds are a linux-specific concept)
test [ui] tests\ui\command\command-exec.rs ... ignored, ignored when the operative system is windows (this is a unix-specific test)
test [ui] tests\ui\command\command-pre-exec.rs#mir ... ignored, ignored when the operative system is windows (this is a unix-specific test)
test [ui] tests\ui\command\command-pre-exec.rs#thir ... ignored, ignored when the operative system is windows (this is a unix-specific test)
test [ui] tests\ui\command\command-setgroups.rs ... ignored, ignored when the operative system is windows (this is a unix-specific test)
---
test [ui] tests\ui\issues\issue-18423.rs ... ok
test [ui] tests\ui\issues\issue-18119.rs ... ok
test [ui] tests\ui\issues\issue-18685.rs ... ok
test [ui] tests\ui\issues\issue-18576.rs ... ok
test [ui] tests\ui\issues\issue-18804\main.rs ... ignored, ignored when the operative system is windows (no extern_weak linkage)
test [ui] tests\ui\issues\issue-18738.rs ... ok
test [ui] tests\ui\issues\issue-18514.rs ... ok
test [ui] tests\ui\issues\issue-18767.rs ... ok
test [ui] tests\ui\issues\issue-18906.rs ... ok
---
test [ui] tests\ui\issues\issue-6919.rs ... ok
test [ui] tests\ui\issues\issue-69396-const-no-type-in-macro.rs ... ok
test [ui] tests\ui\issues\issue-69225-SCEVAddExpr-wrap-flag.rs ... ok
test [ui] tests\ui\issues\issue-69225-layout-repeated-checked-add.rs ... ok
test [ui] tests\ui\issues\issue-70093\issue-70093-link-directives.rs ... ignored, ignored when the operative system is windows (this will probably only work on unixish systems)
test [ui] tests\ui\issues\issue-70093\issue-70093.rs ... ignored, ignored when the operative system is windows (this will probably only work on unixish systems)
test [ui] tests\ui\issues\issue-66923-show-error-for-correct-call.rs ... ok
test [ui] tests\ui\issues\issue-70381.rs ... ok
test [ui] tests\ui\issues\issue-7044.rs ... ok
test [ui] tests\ui\issues\issue-70673.rs ... ok
---
test [ui] tests\ui\overloaded\overloaded_deref_with_ref_pattern.rs ... ok
test [ui] tests\ui\overloaded\overloaded-index-assoc-list.rs ... ok
test [ui] tests\ui\overloaded\overloaded-index.rs ... ok
test [ui] tests\ui\overloaded\overloaded_deref_with_ref_pattern_issue15609.rs ... ok
test [ui] tests\ui\packed\packed-struct-borrow-element-64bit.rs ... ignored, ignored when the pointer width is 32bit ((needs `usize` to be 8-aligned to reproduce all the errors below))
test [ui] tests\ui\packed\packed-struct-borrow-element.rs ... ok
test [ui] tests\ui\packed-struct\packed-struct-transmute.rs ... ok
test [ui] tests\ui\overloaded\overloaded-deref-count.rs ... ok
test [ui] tests\ui\packed\issue-27060-2.rs ... ok
---
---- [ui] tests\ui\symbol-names\x86-stdcall.rs stdout ----

error: test compilation failed although it shouldn't!
status: exit code: 1
command: PATH="C:\a\rust\rust\build\i686-pc-windows-gnu\stage2\bin;C:\a\rust\rust\build\i686-pc-windows-gnu\stage0-bootstrap-tools\i686-pc-windows-gnu\release\deps;C:\a\rust\rust\build\i686-pc-windows-gnu\stage0\bin;C:\a\rust\rust\ninja;C:\a\rust\rust\mingw32\bin;C:\hostedtoolcache\windows\Python\3.11.2\x64\Scripts;C:\hostedtoolcache\windows\Python\3.11.2\x64;C:\msys64\usr\bin;C:\a\rust\rust\sccache;C:\Program Files\MongoDB\Server\5.0\bin;C:\aliyun-cli;C:\vcpkg;C:\cf-cli;C:\Program Files (x86)\NSIS;C:\tools\zstd;C:\Program Files\Mercurial;C:\hostedtoolcache\windows\stack\2.9.3\x64;C:\cabal\bin;C:\ghcup\bin;C:\Program Files\dotnet;C:\mysql\bin;C:\Program Files\R\R-4.2.2\bin\x64;C:\SeleniumWebDrivers\GeckoDriver;C:\Program Files (x86)\sbt\bin;C:\Program Files (x86)\GitHub CLI;C:\Program Files\Git\bin;C:\Program Files (x86)\pipx_bin;C:\npm\prefix;C:\hostedtoolcache\windows\go\1.17.13\x64\bin;C:\hostedtoolcache\windows\Python\3.7.9\x64\Scripts;C:\hostedtoolcache\windows\Python\3.7.9\x64;C:\hostedtoolcache\windows\Ruby\2.5.9\x64\bin;C:\tools\kotlinc\bin;C:\hostedtoolcache\windows\Java_Temurin-Hotspot_jdk\8.0.362-9\x64\bin;C:\Program Files\ImageMagick-7.1.1-Q16-HDRI;C:\Program Files (x86)\Microsoft SDKs\Azure\CLI2\wbin;C:\ProgramData\kind;C:\Program Files\Eclipse Foundation\jdk-8.0.302.8-hotspot\bin;C:\Windows\system32;C:\Windows;C:\Windows\System32\Wbem;C:\Windows\System32\WindowsPowerShell\v1.0;C:\Windows\System32\OpenSSH;C:\ProgramData\Chocolatey\bin;C:\Program Files\PowerShell\7;C:\Program Files\Microsoft\Web Platform Installer;C:\Program Files\Microsoft SQL Server\130\Tools\Binn;C:\Program Files\Microsoft SQL Server\Client SDK\ODBC\170\Tools\Binn;C:\Program Files (x86)\Windows Kits\10\Windows Performance Toolkit;C:\Program Files (x86)\Microsoft SQL Server\110\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\120\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\130\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\140\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\150\DTS\Binn;C:\Program Files (x86)\Microsoft SQL Server\160\DTS\Binn;C:\Program Files\OpenSSL\bin;C:\Strawberry\c\bin;C:\Strawberry\perl\site\bin;C:\Strawberry\perl\bin;C:\ProgramData\chocolatey\lib\pulumi\tools\Pulumi\bin;C:\Program Files\TortoiseSVN\bin;C:\Program Files\CMake\bin;C:\ProgramData\chocolatey\lib\maven\apache-maven-3.8.7\bin;C:\Program Files\Microsoft Service Fabric\bin\Fabric\Fabric.Code;C:\Program Files\Microsoft SDKs\Service Fabric\Tools\ServiceFabricLocalClusterManager;C:\Program Files\nodejs;C:\Program Files\Git\cmd;C:\Program Files\Git\mingw64\bin;C:\Program Files\Git\usr\bin;C:\Program Files\GitHub CLI;C:\tools\php;C:\Program Files (x86)\sbt\bin;C:\SeleniumWebDrivers\ChromeDriver;C:\SeleniumWebDrivers\EdgeDriver;C:\Program Files\Amazon\AWSCLIV2;C:\Program Files\Amazon\SessionManagerPlugin\bin;C:\Program Files\Amazon\AWSSAMCLI\bin;C:\Program Files (x86)\Google\Cloud SDK\google-cloud-sdk\bin;C:\Program Files (x86)\Microsoft BizTalk Server;C:\Program Files\LLVM\bin;C:\Users\runneradmin\.dotnet\tools;C:\Users\runneradmin\.cargo\bin;C:\Users\runneradmin\AppData\Local\Microsoft\WindowsApps" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage2\\bin\\rustc.exe" "C:\\a\\rust\\rust\\tests\\ui\\symbol-names\\x86-stdcall.rs" "-Zthreads=1" "--target=i686-pc-windows-gnu" "--error-format" "json" "--json" "future-incompat" "-Ccodegen-units=1" "-Zui-testing" "-Zsimulate-remapped-rust-src-base=/rustc/FAKE_PREFIX" "-Ztranslate-remapped-path-to-local-path=no" "-Zdeduplicate-diagnostics=no" "-Cstrip=debuginfo" "--remap-path-prefix=C:\\a\\rust\\rust\\tests\\ui=fake-test-src-base" "-C" "prefer-dynamic" "--out-dir" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\test\\ui\\symbol-names\\x86-stdcall" "-A" "unused" "-Crpath" "-Cdebuginfo=0" "-Lnative=C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\native\\rust-test-helpers" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\test\\ui\\symbol-names\\x86-stdcall\\auxiliary"
stdout: none
--- stderr -------------------------------
error: linking with `i686-w64-mingw32-gcc` failed: exit code: 1
   |
   = note: "i686-w64-mingw32-gcc" "-Wl,C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcNCFKA7\\list.def" "-fno-use-linker-plugin" "-Wl,--dynamicbase" "-Wl,--disable-auto-image-base" "-Wl,--large-address-aware" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage2\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\rsbegin.o" "C:\\Users\\RUNNER~1\\AppData\\Local\\Temp\\rustcNCFKA7\\symbols.o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\test\\ui\\symbol-names\\x86-stdcall\\x86-stdcall.x86_stdcall.69a031ae-cgu.0.rcgu.o" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\native\\rust-test-helpers" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\test\\ui\\symbol-names\\x86-stdcall\\auxiliary" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage2\\lib\\rustlib\\i686-pc-windows-gnu\\lib" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage2\\lib\\rustlib\\i686-pc-windows-gnu\\lib" "-lstd-f8ce1d6d5fcac5f3" "-Wl,-Bstatic" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage2\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\libcompiler_builtins-59dfbfb39a743cbc.rlib" "-Wl,-Bdynamic" "-lkernel32" "-ladvapi32" "-luserenv" "-lkernel32" "-lws2_32" "-lbcrypt" "-lntdll" "-lgcc_s" "-lmsvcrt" "-lmingwex" "-lmingw32" "-lgcc" "-lmsvcrt" "-luser32" "-lkernel32" "-Wl,--nxcompat" "-L" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage2\\lib\\rustlib\\i686-pc-windows-gnu\\lib" "-o" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\test\\ui\\symbol-names\\x86-stdcall\\x86_stdcall.dll" "-Wl,--gc-sections" "-shared" "-Wl,--out-implib=C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\test\\ui\\symbol-names\\x86-stdcall\\libx86_stdcall.dll.a" "-Wl,--strip-debug" "-nodefaultlibs" "C:\\a\\rust\\rust\\build\\i686-pc-windows-gnu\\stage2\\lib\\rustlib\\i686-pc-windows-gnu\\lib\\rsend.o"
   = note: C:/a/rust/rust/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: warning: resolving _bar by linking to @bar@4
           Use --enable-stdcall-fixup to disable these warnings
           Use --disable-stdcall-fixup to disable these fixups
           C:/a/rust/rust/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: warning: resolving _foo by linking to _foo@4
           C:/a/rust/rust/mingw32/bin/../lib/gcc/i686-w64-mingw32/12.2.0/../../../../i686-w64-mingw32/bin/ld.exe: cannot export baz: symbol not defined
           collect2.exe: error: ld returned 1 exit status

error: aborting due to previous error
------------------------------------------

---
test result: FAILED. 14506 passed; 1 failed; 227 ignored; 0 measured; 0 filtered out; finished in 939.55s

Some tests failed in compiletest suite=ui mode=ui host=i686-pc-windows-gnu target=i686-pc-windows-gnu
Build completed unsuccessfully in 0:57:34
make: *** [Makefile:80: ci-mingw-subset-2] Error 1
```

</details>

bors commented:
:lock: Merge conflict

This pull request and the master branch diverged in a way that cannot be automatically merged. Please rebase on top of the latest master branch, and let the reviewer approve again.

<details><summary>How do I rebase?</summary>

Assuming `self` is your fork and `upstream` is this repository, you can resolve the conflict following these steps:

1. `git checkout pa-compiletest-ignore` *(switch to your branch)*
2. `git fetch upstream master` *(retrieve the latest master)*
3. `git rebase upstream/master -p` *(rebase on top of it)*
4. Follow the on-screen instruction to resolve conflicts (check `git status` if you got lost).
5. `git push self pa-compiletest-ignore --force-with-lease` *(update this PR)*

You may also read [*Git Rebasing to Resolve Conflicts* by Drew Blessing](http://blessing.io/git/git-rebase/open-source/2015/08/23/git-rebasing-to-resolve-conflicts.html) for a short tutorial.

Please avoid the ["**Resolve conflicts**" button](https://help.github.com/articles/resolving-a-merge-conflict-on-github/) on GitHub. It uses `git merge` instead of `git rebase` which makes the PR commit history more difficult to read.

Sometimes step 4 will complete without asking for resolution. This is usually due to difference between how `Cargo.lock` conflict is handled during merge and rebase. This is normal, and you should still perform step 5 to update this PR.

</details>

<details><summary>Error message</summary>

```text
Auto-merging tests/codegen/issues/issue-37945.rs
CONFLICT (content): Merge conflict in tests/codegen/issues/issue-37945.rs
Auto-merging src/tools/compiletest/src/main.rs
CONFLICT (content): Merge conflict in src/tools/compiletest/src/main.rs
Auto-merging compiler/rustc_driver_impl/src/lib.rs
Automatic merge failed; fix conflicts and then commit the result.

```

</details>

pietroalbini commented:
@bors r=ehuss rollup=never

bors commented:
:pushpin: Commit ff7045df63749a6d90aafc59f4fdc3cc915d332b has been approved by `ehuss`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ehuss ff7045df63749a6d90aafc59f4fdc3cc915d332b -->
<!-- homu: {"type":"Approved","sha":"ff7045df63749a6d90aafc59f4fdc3cc915d332b","approver":"ehuss","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

rust-log-analyzer commented:
The job **`x86_64-gnu-llvm-14`** failed! Check out the build log: [(web)](https://github.com/rust-lang/rust/actions/runs/4594067748/jobs/8112656250) [(plain)](https://github.com/rust-lang/rust/commit/ff7045df63749a6d90aafc59f4fdc3cc915d332b/checks/12468511268/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
Prepare all required actions
Getting action download info
Download action repository 'actions/checkout@v3' (SHA:8f4b7f84864484a7bf31766abe9204da3cbe65b3)
Download action repository 'rust-lang/simpleinfra@master' (SHA:3040325909b538d8ad81ad89a04b7a91b109c313)
Complete job name: PR (x86_64-gnu-llvm-14, false, ubuntu-20.04-16core-64gb)
git config --global core.autocrlf false
shell: /usr/bin/bash --noprofile --norc -e -o pipefail {0}
env:
  CI_JOB_NAME: x86_64-gnu-llvm-14
---
test result: ok. 236 passed; 0 failed; 8 ignored; 0 measured; 0 filtered out; finished in 14.64s

 finished in 14.714 seconds
Check compiletest suite=codegen mode=codegen (x86_64-unknown-linux-gnu -> x86_64-unknown-linux-gnu)
thread 'main' panicked at 'invalid line in /checkout/tests/codegen/issues/issue-37945.rs: ignore-32-bit LLVM has a bug with them', src/tools/compiletest/src/header.rs:944:42
Build completed unsuccessfully in 0:11:48
```

</details>

pietroalbini commented:
@bors r=ehuss rollup=never

bors commented:
:bulb: This pull request was already approved, no need to approve it again.

- There's another pull request that is currently being tested, blocking this pull request: #109884

bors commented:
:pushpin: Commit ff7045df63749a6d90aafc59f4fdc3cc915d332b has been approved by `ehuss`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ehuss ff7045df63749a6d90aafc59f4fdc3cc915d332b -->
<!-- homu: {"type":"Approved","sha":"ff7045df63749a6d90aafc59f4fdc3cc915d332b","approver":"ehuss","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

pietroalbini commented:
@bors r=ehuss rollup=never

bors commented:
:pushpin: Commit bbcbb6fccee0bc9edd7f4da96426d33a1742687d has been approved by `ehuss`

It is now in the [queue](https://bors.rust-lang.org/homu/queue/rust) for this repository.

<!-- @bors r=ehuss bbcbb6fccee0bc9edd7f4da96426d33a1742687d -->
<!-- homu: {"type":"Approved","sha":"bbcbb6fccee0bc9edd7f4da96426d33a1742687d","approver":"ehuss","queue":"https://bors.rust-lang.org/homu/queue/rust"} -->

bors commented:
:broken_heart: Test failed - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4595388165/jobs/8116120584)
<!-- homu: {"type":"BuildFailed","builder_url":"https://github.com/rust-lang-ci/rust/actions/runs/4595388165/jobs/8116120584","builder_name":"checks-actions"} -->

rust-log-analyzer commented:
The job **`armhf-gnu`** failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4595388165/jobs/8115593337) [(plain)](https://github.com/rust-lang-ci/rust/commit/d9c7e74b0e033d90a01f96ac90ff14d5380f182e/checks/12472258418/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain
test [ui] tests/ui/symbol-names/impl1.rs#legacy ... ok
test [ui] tests/ui/symbol-names/issue-53912.rs ... ok
test [ui] tests/ui/symbol-names/issue-60925.rs#legacy ... ok
test [ui] tests/ui/symbol-names/issue-76365.rs#v0 ... ok
test [ui] tests/ui/symbol-names/x86-stdcall.rs ... ignored, ignored when the target environment is gnu (vectorcall is not supported by GCC: https://gcc.gnu.org/bugzilla/show_bug.cgi?id=89485)
test [ui] tests/ui/symbol-names/issue-60925.rs#v0 ... ok
test [ui] tests/ui/symbol-names/trait-objects.rs#v0 ... ok
test [ui] tests/ui/symbol-names/issue-75326.rs#legacy ... ok
test [ui] tests/ui/symbol-names/issue-75326.rs#v0 ... ok
---
---- [ui] tests/ui/associated-consts/associated-const-use-default.rs stdout ----

error: test run failed!
status: exit status: 101
command: RUST_TEST_THREADS="16" "/checkout/obj/build/x86_64-unknown-linux-gnu/stage0-tools-bin/remote-test-client" "run" "0" "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/associated-consts/associated-const-use-default/a"
uploaded "/checkout/obj/build/x86_64-unknown-linux-gnu/test/ui/associated-consts/associated-const-use-default/a", waiting for result
------------------------------------------
--- stderr -------------------------------
thread 'main' panicked at 'client.read_exact(&mut header) failed with Connection reset by peer (os error 104)', src/tools/remote-test-client/src/main.rs:310:9
```

</details>

pietroalbini commented:
@bors retry

bors commented:
:hourglass: Testing commit bbcbb6fccee0bc9edd7f4da96426d33a1742687d with merge b728851503dafba95befe707fee2ec3c1a553949...
<!-- homu: {"type":"BuildStarted","head_sha":"bbcbb6fccee0bc9edd7f4da96426d33a1742687d","merge_sha":"b728851503dafba95befe707fee2ec3c1a553949"} -->

rust-log-analyzer commented:
A job failed! Check out the build log: [(web)](https://github.com/rust-lang-ci/rust/actions/runs/4598728843/jobs/8123175872) [(plain)](https://github.com/rust-lang-ci/rust/commit/b728851503dafba95befe707fee2ec3c1a553949/checks/12482185416/logs)

<details><summary><i>Click to see the possible cause of the failure (guessed by this bot)</i></summary>

```plain

```

</details>

bors commented:
:hourglass: Testing commit bbcbb6fccee0bc9edd7f4da96426d33a1742687d with merge b2b676d886ded452125712a304c524c08360f916...
<!-- homu: {"type":"BuildStarted","head_sha":"bbcbb6fccee0bc9edd7f4da96426d33a1742687d","merge_sha":"b2b676d886ded452125712a304c524c08360f916"} -->

scottmcm commented:
Thanks for making this?  I literally just yesterday caught myself making a type in `// ignore-`!

bors commented:
:sunny: Test successful - [checks-actions](https://github.com/rust-lang-ci/rust/actions/runs/4620593912/jobs/8173466122)
Approved by: ehuss
Pushing b2b676d886ded452125712a304c524c08360f916 to master...
<!-- homu: {"type":"BuildCompleted","approved_by":"ehuss","base_ref":"master","builders":{"checks-actions":"https://github.com/rust-lang-ci/rust/actions/runs/4620593912/jobs/8173466122"},"merge_sha":"b2b676d886ded452125712a304c524c08360f916"} -->

rust-timer commented:
Finished benchmarking commit (b2b676d886ded452125712a304c524c08360f916): [comparison URL](https://perf.rust-lang.org/compare.html?start=4cb92cc83ac6b24d83f8a30763b276cadb8dc41d&end=b2b676d886ded452125712a304c524c08360f916&stat=instructions:u).

### Overall result: ✅ improvements - no action needed


@rustbot label: -perf-regression



### [Instruction count](https://perf.rust-lang.org/compare.html?start=4cb92cc83ac6b24d83f8a30763b276cadb8dc41d&end=b2b676d886ded452125712a304c524c08360f916&stat=instructions:u)
This is a highly reliable metric that was used to determine the overall result at the top of this comment.

|                                    | mean  | range          | count |
|:----------------------------------:|:-----:|:--------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -     | -              | 0     |
| Regressions ❌ <br /> (secondary)  | -     | -              | 0     |
| Improvements ✅ <br /> (primary)   | -     | -              | 0     |
| Improvements ✅ <br /> (secondary) | -2.0% | [-2.0%, -2.0%] | 1     |
| All ❌✅ (primary)                 | -     | -              | 0     |

### [Max RSS (memory usage)](https://perf.rust-lang.org/compare.html?start=4cb92cc83ac6b24d83f8a30763b276cadb8dc41d&end=b2b676d886ded452125712a304c524c08360f916&stat=max-rss)
<details>
<summary>Results</summary>

This is a less reliable metric that may be of interest but was not used to determine the overall result at the top of this comment.

|                                    | mean | range        | count |
|:----------------------------------:|:----:|:------------:|:-----:|
| Regressions ❌ <br /> (primary)    | -    | -            | 0     |
| Regressions ❌ <br /> (secondary)  | 2.9% | [1.5%, 4.4%] | 4     |
| Improvements ✅ <br /> (primary)   | -    | -            | 0     |
| Improvements ✅ <br /> (secondary) | -    | -            | 0     |
| All ❌✅ (primary)                 | -    | -            | 0     |
</details>

### [Cycles](https://perf.rust-lang.org/compare.html?start=4cb92cc83ac6b24d83f8a30763b276cadb8dc41d&end=b2b676d886ded452125712a304c524c08360f916&stat=cycles:u)
This benchmark run did not return any relevant results for this metric.
